Fix bug that the end stone smelter does not open the menu.

uspiritwolf opened the pull request:
Fixed bug that EndStoneSmelter doesn't work, i.e. it's not possible to open the menu and use it. 

In 1.21 there is no old "use" method anymore, so new "useWithoutItem" is implemented instead of the old one.

uspiritwolf committed:
Fix the end stone smelter does not open the menu.

OpenBagTwo commented:
Confirming that this is working, though it looks like Minecraft is now expecting a recipe book icon in the GUI.

![End stone smelter smelting away](https://github.com/user-attachments/assets/d7675736-e107-4450-ac61-06a7e5810a3a)

uspiritwolf committed:
Fixed recipe book icon in the end stone smelter screen

uspiritwolf commented:
Updated code: Fixed the recipe book icon in the GUI.

![2024-08-05_18 11 51](https://github.com/user-attachments/assets/35583beb-2393-4dd3-942a-4e30645b4055)

OpenBagTwo commented:
Confirmed on my end. Looks great!

Sarik200809 commented:
Can I just download this fix as a .jar file? I tried to build it with source, but it just gave errors about dependences.
Thanks.

OpenBagTwo commented:
@anit24 I've published a build [here](https://github.com/OpenBagTwo/BetterEnd/releases/tag/21.0.9%2Bopenbagtwo.24.8.10) that also contains some additional fixes. It should work as a drop-in replacement for BetterEnd 21.0.8 (read: use the latest WorldWeaver and BCLib). **Please** be cautious using the mod (make lots of backups), and if you have any issues, do not report them to this project unless you have confirmed the issues are also present in the official releases.

OpenBagTwo commented:
Hey @uspiritwolf it looks like Frank went ahead and resolved this himself with 3593c077d3aff0eecf11290c07716e0443c3923e and ffe9ebe

uspiritwolf commented:
Well, then I'm closing the pull request

OpenBagTwo commented:
> Well, then I'm closing the pull request

Thank you so much for making it in the first place—it meant a week of being able to alloy while we waited for the official fix.
